sdk/identity/identity/samples/AzureIdentityExamples.md

```diff
@@ -908,6 +915,153 @@ class RotatingCertificateCredential implements TokenCredential {
+
+In many cases applications require tight control over user interaction. In these applications automatically blocking on required user interaction is often undesired or impractical. For this reason, credentials in the @azure/identity library which interact with the user offer mechanisms to fully control user interaction.
+
+```
+  const options: InteractiveBrowserCredentialOptions = {
+    disableAutomaticAuthentication: true
+  };
+  const credential = new InteractiveBrowserCredential(options);
+  await credential.authenticate("https://vault.azure.net/.default");
+  const client = new SecretClient("https://key-vault-name.vault.azure.net", credential);
+```
+
+In this sample the application is again using the `InteractiveBrowserCredential` to authenticate a SecretClient, but with two major differences from our first example. First, in this example the application is explicitly forcing any user interaction to happen before the credential is given to the client by calling `authenticate`.
+
+The second difference is here the application is preventing the credential from automatically initiating user interaction. Even though the application authenticates the user before the credential is used, further interaction might still be needed, for instance in the case that the user's refresh token expires, or a specific method require additional consent or authentication.
+
+By setting the option `disableAutomaticAuthentication` to `true` the credential will fail to automatically authenticate calls where user interaction is necessary. Instead, the credential will throw an error `AuthenticationRequiredError`. The following example demonstrates an application handling such an exception to prompt the user to authenticate only after some application logic has completed.
+
+```
+try
+{
+    await client.getSecret("secret-name");
+}
+catch (e)
+{
+  await ensureAnimationCompleted();
+
+  if(e instanceof AuthenticationRequiredError){
+    await credential.authenticate(e.scopes);
+    client.getSecret("secret-name");
+  }
+  else{
+    throw (e);
+  }
+}
+```
+
+### Persisting user authentication data
+
+Quite often applications desire the ability to be run multiple times without having to re-authenticate the user on each execution. This requires that data from credentials be persisted outside of the application memory so that it can authenticate silently on subsequent executions. Applications can persist this data using `tokenCachePersistenceOptions` when constructing the credential, and persisting the `authenticationRecord` returned from `authenticate`.
+
+#### Persisting the token cache
+
+The credential handles persisting all the data needed to silently authenticate one or many accounts. It manages sensitive data such as refresh tokens and access tokens which must be protected to prevent compromising the accounts related to them. By default, the `@azure/identity` library will protect and cache sensitive token data using available platform data protection.
```

Comment:
```suggestion
The credential handles persisting all the data needed to authenticate one or many accounts silently. It manages sensitive data such as refresh tokens and access tokens which must be protected to prevent compromising the accounts related to them. By default, the `@azure/identity` library will protect and cache sensitive token data using available platform data protection.
```